File: problems/535/solution.py
class Codec:

    def encode(self, longUrl: str) -> str:
        """Encodes a URL to a shortened URL.
        """
        s = ''

        for ch in longUrl:
            s += str(ord(ch)) + '-'
        
        return s
        

    def decode(self, shortUrl: str) -> str:
        """Decodes a shortened URL to its original URL.
        """
        
        return ''.join(map(lambda n: chr(int(n)) if n else '', shortUrl.split('-')))
        

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(url))

File: problems/535/solution.cpp
class Solution {
private:
    hash<string> hashString;
    unordered_map<unsigned long, string> store;

public:

    // Encodes a URL to a shortened URL.
    string encode(string longUrl) {
        unsigned long hash = hashString(longUrl);
        store[hash] = longUrl;
        
        return to_string(hash);
    }

    // Decodes a shortened URL to its original URL.
    string decode(string shortUrl) {
        return store[stoul(shortUrl, nullptr, 0)];
    }
};

// Your Solution object will be instantiated and called as such:
// Solution solution;
// solution.decode(solution.encode(url));